### ISSUE TITLE
support utmp records files

### ISSUE THREAD

**Issue by jtmoon79** [jtmoon79/super-speedy-syslog-searcher#70]
### Problem

Custom binary log files `wtmp`, `utmp`, `btmp`, etc. are common. Not being able to process a `*tmp` file loses important logged information (which degrades the utility of this project).

### Solution

Support processing `*tmp` files.  They are remarkably different than processing ad-hoc syslog files.

A "wtmp processor class" would be adjacent to a `SyslogProcessor`. In other words, all functionality of a `SyslogProcessor` (and it's contained `SyslineReader`, `LineReader`, `BlockReader`) would be created anew in some `WtmpProcessor` "class".

Looking at the ordering of wtmp entries:
```text
$ last -i -w -x --time-format 'iso' -f /var/log/wtmp | head -n 100
root     pts/0        131.191.30.152   2022-12-28T00:40:37-08:00 - 2022-12-28T00:40:40-08:00  (00:00)
root     pts/1        131.191.30.152   2022-12-28T00:40:06-08:00   still logged in
root     pts/1        131.191.30.152   2022-12-28T00:39:28-08:00 - 2022-12-28T00:39:47-08:00  (00:00)
root     pts/0        131.191.30.152   2022-12-27T23:07:29-08:00 - 2022-12-28T00:40:15-08:00  (01:32)
root     pts/0        131.191.30.152   2022-12-26T10:23:54-08:00 - 2022-12-26T12:06:19-08:00  (01:42)
root     pts/0        131.191.30.152   2022-12-25T18:19:58-08:00 - 2022-12-26T09:27:51-08:00  (15:07)
```

First, there are two datetime stamps per entry. That means for `s4` there are two datetime events created per wtmp entry.
Second, it appears the entries are ordered by the first datetimestamp of the entry (login time). The second datetimestamp (logout time) may be any time after the first (and is not ordered relative to other entries).

The implication is only datetime filter `--dt-before` can be used for a binary search of entries. But once the nearest entry for the `--dt-before` datetimestamp is found, _all_ prior entries must be parsed, no matter the value of `--dt-after`.

<br/>

Here is [a perl technique for parsing `*tmp` files](https://www.hcidata.info/wtmp.htm) ([archived](https://archive.vn/YIrl8)).
Here is [OpenGroup definitions for `utmpx.h`](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/utmpx.h.html)

**Comment by jtmoon79** [comment]
Similar to #17

**Comment by jtmoon79** [comment]
A hack first implementation would be to call program `last` and process the stdout.

Sample output:

```text
last -a -i -w -x --time-format 'iso' -f /var/log/wtmp
root     pts/0        2022-12-19T23:24:43-08:00   still logged in                        174.61.163.8
root     pts/0        2022-12-19T23:09:48-08:00 - 2022-12-19T23:17:22-08:00  (00:07)     174.61.163.8
root     pts/2        2022-12-19T22:35:16-08:00 - 2022-12-19T22:36:00-08:00  (00:00)     0.0.0.0
root     pts/2        2022-12-19T22:35:13-08:00 - 2022-12-19T22:35:16-08:00  (00:00)     0.0.0.0
root     pts/2        2022-12-19T22:35:13-08:00 - 2022-12-19T22:35:13-08:00  (00:00)     0.0.0.0
root     pts/2        2022-12-19T22:34:49-08:00 - 2022-12-19T22:35:13-08:00  (00:00)     0.0.0.0
```

This would be a good first step to adding the processing support relatively easy.

**Comment by jtmoon79** [comment]
Fixed in b227f531a6f348cdd9b3fa5fe010adf979dd8e98 b67a84018152dca28881bbc8f62e877b27b12b8d 00844f61d76709971f879a6ef9d0c1f3be1d4387 40d3538a158863d518763fdb07ce48754e5feb81 7282788d9d2ba103fd8e5186890699d7fa17f6ad
